### ISSUE TITLE
Wrong RTL (Persian/Arabic) two digit number rendering

### ISSUE THREAD

**Issue by molaeiali** [osmandapp/OsmAnd#19542]
### Description

Two digits numbers in RTL languages (at least Persian numbers) are being rendered wrongly, for example, the numbers 16, 18 and 20 (۱۶، ۱۸، ۲۰) are being shown as 61, 81, and 02 (۶۱، ۸۱، ۰۲)

In Persian and Arabic, the text is written Right-to-Left but the digits should still be Left-to-Right.
![photo_2024-04-13_14-34-38](https://github.com/osmandapp/OsmAnd/assets/12999682/0c4cb848-bc2c-4c63-baa8-24c227ea1019)

I've checked openstreetmap.org website and the numbers are correct there, so I think it's a rendering issue in osmand

![Screenshot from 2024-04-13 14-38-34](https://github.com/osmandapp/OsmAnd/assets/12999682/3dedd288-29fd-4345-84b8-36038deac554)

### Steps to reproduce

Go here on the map as an example:
https://www.openstreetmap.org/#map=19/29.61697/52.48079

### Actual result

Numbers being shown RTL, for example 18 is being shown as 81

### Expected result

In RTL languages numbers should still be LTR so ۱۸/18 is expected not ۸۱/81

### Your Environment (required)

```
OsmAnd Version: OsmAnd~ 4.6.12
Android/iOS version: 13
Device model: Redmi Note 9 - Pixel Experience 13.0-20221113-1311-Official
```

**Comment by Andrewschenko** [comment]
#18928

**Comment by Andrewschenko** [comment]
Should be fixed in 4.7.2

**Comment by molaeiali** [comment]
This issus is somewhat fixed but not fully in OSMAnd~ 4.7.10
![photo_2024-04-27_13-24-38](https://github.com/osmandapp/OsmAnd/assets/12999682/9f6a883e-6e03-41d6-ae4b-552e040a35a0)

As you can see, now if a number is between two words, they are being shown in wrong order,
Correct:
کوچه ۱۷ فلاحی
Currently shown:
فلاحی ۱۷ کوچه

I think in your testing in #18928 you used same letter "س" before and after the number and it caused it to be missed

**Comment by vshcherb** [comment]
Direction should be detected by first word (ignoring digits) or by first character LRM https://en.wikipedia.org/wiki/Left-to-right_mark
